pkg/utils/common/proxy_plugin.go

```diff
@@ -0,0 +1,75 @@
+package common
+
+import (
+	"context"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// CreateProxyPlugin creates an object of ProxyPlugin for the OpenShift route target
+func (s *SuiteController) CreateProxyPlugin(proxyPluginName, proxyPluginNamespace, routeName, routeNamespace string) (*toolchainv1alpha1.ProxyPlugin, error) {
+
+	// Create the ProxyPlugin object
+	proxyPlugin := NewProxyPlugin(proxyPluginName, proxyPluginNamespace, routeName, routeNamespace)
+
+	if err := s.KubeRest().Create(context.TODO(), proxyPlugin); err != nil {
+		return nil, err
+	}
+	return proxyPlugin, nil
+}
+
+// DeleteProxyPlugin deletes the ProxyPlugin object
+func (s *SuiteController) DeleteProxyPlugin(proxyPluginName, proxyPluginNamespace string) (bool, error) {
+	proxyPlugin := &toolchainv1alpha1.ProxyPlugin{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      proxyPluginName,
+			Namespace: proxyPluginNamespace,
+		},
+	}
+
+	if err := s.KubeRest().Delete(context.TODO(), proxyPlugin); err != nil {
+		return false, err
+	}
+	err := utils.WaitUntil(func() (done bool, err error) {
+		err = s.KubeRest().Get(context.TODO(), types.NamespacedName{
+			Namespace: proxyPluginNamespace,
+			Name:      proxyPluginName,
+		}, proxyPlugin)
+
+		if err != nil {
+			if k8sErrors.IsNotFound(err) {
+				return true, nil
+			}
+			return false, err
```

Comment:
need to log the error here, as it will get masked/lost by the timeout error generated if you keep getting errors for the entire 5 minutes